tests/lib/util/source-code.js

```diff
@@ -892,15 +891,22 @@ describe("SourceCode", () => {
 
             const spy = sandbox.spy(assertJSDoc);
 
-            linter.on("FunctionDeclaration", spy);
-            linter.verify(code, { rules: {}, parserOptions: { ecmaVersion: 6 } }, filename, true);
+            linter.defineRule("checker", () => ({ FunctionDeclaration: spy }));
+            linter.verify(code, { rules: { checker: "error" }, parserOptions: { ecmaVersion: 6 } });
             assert.isTrue(spy.calledOnce, "Event handler should be called.");
         });
 
     });
 
     describe("getComments()", () => {
-        const config = { rules: {} };
+        const config = { rules: { checker: "error" }, parserOptions: { sourceType: "module" } };
```

Comment:
Several of the tests in this suite had previously been broken because they weren't setting `sourceType` correctly, so there was a parse error and the rule listener wasn't actually getting called.